### PR TITLE
Windows.System.Display.DisplayRequest

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): #907 

## PR Type
What kind of change does this PR introduce?

 - Feature -->

## What is the current behavior?
`Windows.System.Display.DisplayRequest` is not implemented

## What is the new behavior?
`Windows.System.Display.DisplayRequest` is implemented on iOS and Android. Its behavior matches UWP and it allows creating multiple instances as well as throws when release is attempted when there is no active request.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/System/Display/DisplayRequest.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+		private int _instanceRequestCount = 0;
+
+		public void RequestActive()
+		{
+			lock (_syncLock)
+			{
+				_instanceRequestCount++;
+				_globalRequestCount++;
+				if (_globalRequestCount == 1)
```
Comment (carldebilly) [inline src/Uno.UWP/System/Display/DisplayRequest.cs]: I think there's a way to implement this in a lockless way using `Interlocked.Increment` / `Interlocked.Decrement`

Comment (MartinZikmund) [reply src/Uno.UWP/System/Display/DisplayRequest.cs]: Not sure how to approach this though - as I need to increment/decrement two variables, but the most important thing is that the `ActivateScreenLock` and `DeactivateScreenLock` must run only when the `_globalRequestCount` has an appropriate value. I think I can make sure to safely increment with `Interlocked`, but that won't solve the problem when I check `_globalRequestCount` value (in `if`), enter the block, and at that moment another thread changes the `_globalRequestCount` again, I may for example then deactivate the screen lock although another thread just activated it. Do you have an idea how to resolve this problem?

File: src/Uno.UWP/System/Display/DisplayRequest.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+					ActivateScreenLock();
+				}
+			}
+		}
+
+		public void RequestRelease()
+		{
+			lock (_syncLock)
+			{
+				if (_instanceRequestCount == 0)
```
Comment (jeromelaban) [inline src/Uno.UWP/System/Display/DisplayRequest.cs]: Missing braces

File: src/Uno.UWP/System/Display/DisplayRequest.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+					throw new InvalidOperationException("No active display request to release");
+				_instanceRequestCount--;
+				_globalRequestCount--;
+				if ( _globalRequestCount == 0)
```
Comment (jeromelaban) [inline src/Uno.UWP/System/Display/DisplayRequest.cs]: Formatting

File: src/Uno.UWP/System/Display/DisplayRequest.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UWP/System/Display/DisplayRequest.cs]: Sorry, power of habit, at work we omit braces for single `throw` statements

File: src/Uno.UWP/System/Display/DisplayRequest.cs (hunk 3 above)
Comment (MartinZikmund) [reply src/Uno.UWP/System/Display/DisplayRequest.cs]: Subtle, was confused where the problem is the first time I saw it :-D .

File: src/Uno.UWP/System/Display/DisplayRequest.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UWP/System/Display/DisplayRequest.cs]: I think the `DisplayRequest` class will not commonly be used in high-perf scenarios, most apps will just use it once per launch, for example to keep screen on when playing video, so the cost of the `lock` feels manageable and it is easy to show that the code works correctly as compared to a more complex solution

**Pull request merged**